lib/features/context-pad/ContextPad.js

```diff
@@ -622,25 +630,39 @@ ContextPad.prototype._updatePosition = function() {
  * using the `djs-element-hidden` or `djs-label-hidden` markers.
  */
 ContextPad.prototype._updateVisibility = function() {
-  if (!this.isOpen()) {
+
+  // Only update pad once per frame
+  // cf. https://github.com/camunda/camunda-modeler/issues/4335
+  if (this._requestedFrame) {
```

Comment:
Shall we wrap this into a "scheduling utility"? I forsee that we'll need this in other places, too.

Reply:
I think so. We will have to make it available through dependency injection, since we want to do cancel on destroy centrally as well, which requires the `eventBus`.

Reply:
Alternative: 

* Expose it explicitly as a task to execute; no eventBus and complicated logic needed

```
function MyComponent() {

  // ensure that update is only reset 
  this._updateTask = createTask((some, dependency) => {
    
  });
}

// execute it at any time
myComponent._updateTask.schedule(some, dependency);
```

Reply:
If we delay execution, we will need to be able to cancel it. I want to ensure this is reusable.

When integration testing with bpmn-js, I already ran into issues if we do not properly clean up, as we might try to access DOM elements that have been removed since we scheduled the task. Since this will most likely be relevant to more instances, I would like to have this logic central.

Alternative to dependency injection: make it a utility class and every Component that needs it can create a local Scheduler, with optional cleanup

Reply:
cf. [1e74dff](https://github.com/bpmn-io/diagram-js/pull/915/commits/1e74dff27438d9afb40a97f27d1b3069aeaab275) as a proposed utility